### PR TITLE
Enable batch sender in oltpexporter

### PR DESCRIPTION
**Pull request by sfc-gh-sili**
#### Description

This PR adds opt-in support to oltp exporter for the experimental batch sender (https://github.com/open-telemetry/opentelemetry-collector/issues/8122). By default batch sender is not enabled.

Similar:
* https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34238
* https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32563

#### Link to tracking issue

Resolves https://github.com/open-telemetry/opentelemetry-collector/issues/10834

#### Testing

`exporter/otlpexporter/config_test.go`

#### Documentation

Updated the `oltpexporter` README to point to `exporterhelper` README for batching.

### CODE REVIEW

File: exporter/otlpexporter/factory.go [hunk 1]
```diff
@@ -57,7 +62,9 @@ func createTracesExporter(
 		exporterhelper.WithRetry(oCfg.RetryConfig),
 		exporterhelper.WithQueue(oCfg.QueueConfig),
 		exporterhelper.WithStart(oce.start),
-		exporterhelper.WithShutdown(oce.shutdown))
+		exporterhelper.WithShutdown(oce.shutdown),
+		exporterhelper.WithBatcher(oCfg.BatcherConfig),
```
Comment (dmitryax) [inline exporter/otlpexporter/factory.go:66]: nit: let's move it up a bit to `WithQueue` so all the "sender" options are defined together

File: exporter/otlpexporter/testdata/config.yaml [hunk 2]
```diff
@@ -14,6 +14,11 @@ retry_on_failure:
   multiplier: 1.3
   max_interval: 60s
   max_elapsed_time: 10m
+batcher:
+    enabled: true
+    flush_timeout: 1s
+    min_size_items: 1
+    max_size_items: 10
```
Comment (dmitryax) [inline exporter/otlpexporter/testdata/config.yaml:21]: Just some more realistic values
```suggestion
    flush_timeout: 200ms
    min_size_items: 1000
    max_size_items: 10000
```

**Review by dmitryax — APPROVED** [review]
LGTM. Just few nits

**Comment by dmitryax** [comment]
The failing contrib jobs will be fixed once https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34721 is merged

File: .chloggen/add-batching-option-to-otlp-exporter.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [user]
```
Comment (bogdandrutu) [inline .chloggen/add-batching-option-to-otlp-exporter.yaml:25]: New line please.

File: exporter/otlpexporter/testdata/config.yaml [hunk 4]
```diff
@@ -14,6 +14,11 @@ retry_on_failure:
   multiplier: 1.3
   max_interval: 60s
   max_elapsed_time: 10m
+batcher:
+    enabled: true
+    flush_timeout: 200ms
+    min_size_items: 1000
+    max_size_items: 10000
```
Comment (bogdandrutu) [inline exporter/otlpexporter/testdata/config.yaml:21]: Indentation is a bit off compare to others.

**Review by bogdandrutu — APPROVED** [review]

File: exporter/otlpexporter/testdata/config.yaml (hunk 4 above)
Comment (sfc-gh-sili) [reply exporter/otlpexporter/testdata/config.yaml:21]: Thanks!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**